WordPress/gutenberg
# Global Styles: synchronize user CPT registration and UI visibility

Follow-up to https://github.com/WordPress/gutenberg/pull/34885#discussion_r723393674

This PR centralizes the logic to decide whether to register the user CPT for global styles and whether to make the Global Styles UI visible. It also updates when the user CPT is registered: when the theme has a `theme.json` or block templates, following recent changes in the sidebar.

## Context

The Global Styles sidebar is the tool the user interacts with to provide their own global styles. These styles are stored in a CPT. The registration of the user CPT and the visibility of the GS styles sidebar should go in sync: if the user CPT is not registered the sidebar should not be visible either. We had separate checks in the server and client for this. When we made the GS UI available for all themes that could load the site editor at https://github.com/WordPress/gutenberg/pull/34885 the user CPT was still only registered for themes with `theme.json`, hence we had failures in some themes because the CPT was not registered.

## How to test

Test that it works for a theme with `theme.json` and block templates:

- Activate the empty theme.
- Go to the site editor and make changes to the global styles. They should work fine.

Test that it works for a theme with block templates but not `theme.json`:

- Remove the `theme.json` from the empty theme and reload the site editor.
- The expected result is that the global styles sidebar should still be visible and user changes take effect.

Test that it works for a theme with `theme.json` but not block templates:

- Recover the `theme.json` of the theme.
- Remove the `block-templates/index.html` and reload the site editor.
- The expected result is that you should not be able to load the site editor.
